Fix/jwt auth

iclegane opened the pull request.

Commit:
fix package-lock.json

Commit:
fix README.md

Commit:
fix auth methods

Commit:
added auth check method

Commit:
fix n^2 complexity

Commit:
fix default routes

Commit:
fix default routes

Commit:
separated the web socket logic

Commit:
refactoring vars name

Commit:
fixed init functions

Commit:
added loader indicator

Commit:
loader indicator refactoring

Ayub-Begimkulov commented on server/src/service/WS/WS.ts:
```diff
@@ -85,23 +66,29 @@ export class WSServer {
-        logger.error('Invalid message format')
 
-        return
+      const baseMessage = await WSMessageValidator.validateBaseMessage(data)
+      if (!baseMessage) return
+
+      switch (baseMessage.type) {
+        case 'end': {
+          const endMessage = await WSMessageValidator.validateEndMessage(data)
+          if (endMessage) {
+            await WSMessageHandler.handleCardPosition(
+              endMessage.id,
+              endMessage.position
+            )
+          }
+          break
+        }
```
Получился не совсем правильный вынос, в идеале сделать возможность декларировать различные обработчики снаружи сервиса, а сам сервис отвечает за коннекшен и его поддержание

```ts
class BoardHandlers {
  constructor(wsServer: WSServer) {
    wsServer.on('message', (message) => {
      if (message.type === 'ADD_CARD') {
          this.addCard()
      }
    })
  }

  addCard() {}

  getCards() {
    // get cards
    this.wsServer.send('CARDS', allCards)
  }
}

```

Ayub-Begimkulov reviewed (commented):
- Делать редирект с `/` на `/board` или `/login` в зависимости от страницы 
- Делать редирект на логин после создания аккаунта
- Нормальная обработка ошибок для форм в UI (показывать ошибки под формой)
- WS.socket on message — там может быть не тоько invalid json ошибка, но и любая другая в колбэке
- Нет нормального обновление jwt через рефреш токен) 
- В идеале делать все через сокеты и не иметь рест апишку для карточек (в нашем кейсе)

Commit:
fixed:
Делать редирект с / на /board или /login в зависимости от страницы
Делать редирект на логин после создания аккаунта
Нормальная обработка ошибок для форм в UI (показывать ошибки под формой)
WS.socket on message — там может быть не тоько invalid json ошибка, но и любая другая в колбэке
Нет нормального обновление jwt через рефреш токен)

Commit:
fixed

iclegane committed:
Merge pull request #8 from iclegane/fix/JWT-auth-2

fixed:
